enterprise/backend/src/metabase_enterprise/data_editing/data_editing.clj

```diff
@@ -159,14 +162,22 @@
                                          :when (or before after)]
                                [(diff->pk diff) [before after]])]))))
     ;; table notification system events
-    (doseq [[event payloads] (->> diffs
-                                  (group-by row-update-event)
-                                  (remove (comp #{::no-op} key)))]
+    (doseq [[{single-event :single
+              bulk-event :bulk} payloads] (->> diffs
+                                               (group-by row-update-event)
+                                               (remove (comp #{::no-op} key)))]
       (doseq [[table-id payloads] (group-by :table-id payloads)
-              :let [row-changes (for [{:keys [before after] :as diff} payloads]
+              :let [db-id       (:db-id (first payloads))
```

Comment:
I don't think it's a good idea to assume that all tables are in the same database.

For example a workflow action might modify multiple databases.

Reply:
this is grouping by the table-id right?

Reply:
don't tell me a table can have many dbs :D

Reply:
I missed that `payloads` was being shadowed, so yup this is all fine!